src/tasks/global/spookyraven.ts

```diff
@@ -0,0 +1,117 @@
+import { CombatStrategy, Guards, Task } from "grimoire-kolmafia";
+import { adv1, cliExecute, visitUrl } from "kolmafia";
+import {
+  $effect,
+  $familiar,
+  $item,
+  $location,
+  $monster,
+  $skill,
+  Cartography,
+  Macro,
+  get,
+  have,
+  questStep,
+} from "libram";
+import { getCombat } from "../../lib/combat";
+import { basicEffects, noncombatEffects } from "../../lib/effects";
+import { getEquipment } from "../../lib/equipment";
+import { wandererEquipment } from "./wanderer";
+
+export function spookyravenTasks(): Task[] {
+  return [
+    {
+      name: "Haunted Billiards Room",
+      ready: () => have($item`Spookyraven billiards room key`),
+      completed: () => have($item`[7302]Spookyraven library key`),
+      effects: noncombatEffects,
+      outfit: {
+        equip: getEquipment([
+          $item`June cleaver`,
+          $item`Greatest American Pants`,
+          $item`mafia thumb ring`,
+        ]),
+        modifier: "-combat",
+        familiar: $familiar`Disgeist`,
+      },
+      choices: {
+        900: 2, // Lights Out
+        875: 1, // Hustle the ghost
+        1436: 2, // Maps: go straight to hustle
+      },
+      combat: new CombatStrategy().macro(getCombat(Macro.runaway())),
+      do: $location`The Haunted Billiards Room`,
+    },
+    // Use Jurassic Parka to finish Spookyraven Library, slowly but surely
+    {
+      name: "Spit Acid in Library",
+      ready: () =>
+        have($item`Jurassic Parka`) &&
+        have($item`[7302]Spookyraven library key`) &&
+        questStep("questM20Necklace") < 4 &&
+        get("writingDesksDefeated") < 5,
+      completed: () => have($effect`Everything Looks Yellow`),
```

Comment:
We'll keep adventuring here even after unlocking top floor. Which I guess is fine? I don't think there's anywhere more valuable to YR.

Reply:
We won't because of the quest step check, unless I messed something up